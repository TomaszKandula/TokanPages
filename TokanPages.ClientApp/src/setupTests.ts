// jest-dom adds custom jest matchers for asserting on DOM nodes.
// allows you to do things like:
// expect(element).toHaveTextContent(/react/i)
// learn more: https://github.com/testing-library/jest-dom
import "@testing-library/jest-dom";
import Enzyme from "enzyme";
import Adapter from "enzyme-adapter-react-16";
import enableHooks from "jest-react-hooks-shallow";

enableHooks(jest);

jest.mock("react-redux", () => (
{
    useSelector: jest.fn(),
    useDispatch: jest.fn()
}));

jest.mock("connected-react-router", () => (
{
    connectRouter: jest.fn(),
    routerMiddleware: jest.fn()
}));

Enzyme.configure({ adapter: new Adapter(), disableLifecycleMethods: false });

// environmental variables used during all tests
process.env.REACT_APP_API_VER="1";
process.env.REACT_APP_BACKEND="http://localhost:5000";
