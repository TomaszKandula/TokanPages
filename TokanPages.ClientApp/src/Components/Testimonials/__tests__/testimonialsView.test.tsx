import "../../../setupTests";
import React from "react";
import { shallow } from "enzyme";
import TestimonialsView from "../testimonialsView";

describe("Test component: testimonialsView.", () => 
{
    it("Renders correctly '<TestimonialsView />' when content is loaded.", () => 
    {
        const tree = shallow(<TestimonialsView isLoading={false} content=
        {{
            caption: "Testimonials",
            subtitle: "You can read few commendations...",
            photo1: "joanna.jpg",
            name1: "Joanna",
            occupation1: "Senior Digital Tax Specialist",
            text1: "Joe is very motivated...",
            photo2: "adama.jpg",
            name2: "Adama",
            occupation2: "Full-stack Developer chez DFDS POLSKA",
            text2: "Joe have done very good work...",
            photo3: "scott.jpg",
            name3: "Scott",
            occupation3: "BPO",
            text3: "Joe is one of those rare talents..."
        }}/>);
        expect(tree).toMatchSnapshot();
    });
});
