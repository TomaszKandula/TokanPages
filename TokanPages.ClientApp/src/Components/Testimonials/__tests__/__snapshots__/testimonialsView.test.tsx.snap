// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Test component: testimonialsView. Renders correctly '<TestimonialsView />' when content is loaded. 1`] = `
<section
  className="makeStyles-section-1"
>
  <WithStyles(ForwardRef(Container))
    maxWidth="lg"
  >
    <Styled(MuiBox)
      mb={5}
      pb={10}
      pt={8}
      textAlign="center"
    >
      <WithStyles(ForwardRef(Typography))
        className="makeStyles-caption_text-2"
        data-aos="fade-down"
        gutterBottom={true}
      >
        TESTIMONIALS
      </WithStyles(ForwardRef(Typography))>
    </Styled(MuiBox)>
    <Styled(MuiBox)
      pb={15}
      textAlign="center"
    >
      <WithStyles(ForwardRef(Grid))
        container={true}
        spacing={6}
      >
        <WithStyles(ForwardRef(Grid))
          data-aos="fade-left"
          item={true}
          md={4}
          xs={12}
        >
          <WithStyles(ForwardRef(Card))
            className="makeStyles-card-3"
            elevation={0}
          >
            <WithStyles(ForwardRef(CardMedia))
              className="makeStyles-card_image-4"
              component="img"
              image=" "
            />
            <WithStyles(ForwardRef(CardContent))
              className="makeStyles-card_content-5"
            >
              <WithStyles(ForwardRef(Typography))
                className="makeStyles-card_title-6"
              >
                Joanna
              </WithStyles(ForwardRef(Typography))>
              <WithStyles(ForwardRef(Typography))
                className="makeStyles-card_subheader-7"
              >
                Senior Digital Tax Specialist
              </WithStyles(ForwardRef(Typography))>
              <WithStyles(ForwardRef(Typography))
                className="makeStyles-card_text-8"
              >
                Joe is very motivated...
              </WithStyles(ForwardRef(Typography))>
            </WithStyles(ForwardRef(CardContent))>
          </WithStyles(ForwardRef(Card))>
        </WithStyles(ForwardRef(Grid))>
        <WithStyles(ForwardRef(Grid))
          data-aos="fade-up"
          item={true}
          md={4}
          xs={12}
        >
          <WithStyles(ForwardRef(Card))
            className="makeStyles-card-3"
            elevation={3}
          >
            <WithStyles(ForwardRef(CardMedia))
              className="makeStyles-card_image-4"
              component="img"
              image=" "
            />
            <WithStyles(ForwardRef(CardContent))
              className="makeStyles-card_content-5"
            >
              <WithStyles(ForwardRef(Typography))
                className="makeStyles-card_title-6"
              >
                Adama
              </WithStyles(ForwardRef(Typography))>
              <WithStyles(ForwardRef(Typography))
                className="makeStyles-card_subheader-7"
              >
                Full-stack Developer chez DFDS POLSKA
              </WithStyles(ForwardRef(Typography))>
              <WithStyles(ForwardRef(Typography))
                className="makeStyles-card_text-8"
              >
                Joe have done very good work...
              </WithStyles(ForwardRef(Typography))>
            </WithStyles(ForwardRef(CardContent))>
          </WithStyles(ForwardRef(Card))>
        </WithStyles(ForwardRef(Grid))>
        <WithStyles(ForwardRef(Grid))
          data-aos="fade-right"
          item={true}
          md={4}
          xs={12}
        >
          <WithStyles(ForwardRef(Card))
            className="makeStyles-card-3"
            elevation={3}
          >
            <WithStyles(ForwardRef(CardMedia))
              className="makeStyles-card_image-4"
              component="img"
              image=" "
            />
            <WithStyles(ForwardRef(CardContent))
              className="makeStyles-card_content-5"
            >
              <WithStyles(ForwardRef(Typography))
                className="makeStyles-card_title-6"
              >
                Scott
              </WithStyles(ForwardRef(Typography))>
              <WithStyles(ForwardRef(Typography))
                className="makeStyles-card_subheader-7"
              >
                BPO
              </WithStyles(ForwardRef(Typography))>
              <WithStyles(ForwardRef(Typography))
                className="makeStyles-card_text-8"
              >
                Joe is one of those rare talents...
              </WithStyles(ForwardRef(Typography))>
            </WithStyles(ForwardRef(CardContent))>
          </WithStyles(ForwardRef(Card))>
        </WithStyles(ForwardRef(Grid))>
      </WithStyles(ForwardRef(Grid))>
    </Styled(MuiBox)>
  </WithStyles(ForwardRef(Container))>
</section>
`;
